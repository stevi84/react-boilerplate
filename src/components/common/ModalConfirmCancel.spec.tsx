import { describe, it, expect, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { ModalConfirmCancel } from './ModalConfirmCancel';

describe('ModalConfirmCancel', () => {
  it('should equal saved snapshot', () => {
    const tree = render(
      <ModalConfirmCancel
        isOpen={true}
        headerText={'header'}
        bodyText={'body'}
        confirmButtonText={'confirm'}
        cancelButtonText={'cancel'}
        confirmAction={() => {}}
        cancelAction={() => {}}
      />
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it('should execute confirm when confirmButton is pressed', () => {
    const confirmAction = vi.fn();
    const cancelAction = vi.fn();
    render(
      <ModalConfirmCancel
        isOpen={true}
        headerText={'header'}
        bodyText={'body'}
        confirmButtonText={'confirm'}
        cancelButtonText={'cancel'}
        confirmAction={confirmAction}
        cancelAction={cancelAction}
      />
    );
    fireEvent.click(screen.getByText('confirm'));
    expect(confirmAction.mock.calls.length).toEqual(1);
    expect(cancelAction.mock.calls.length).toEqual(0);
  });

  it('should execute cancel when cancelButton is pressed', () => {
    const confirmAction = vi.fn();
    const cancelAction = vi.fn();
    render(
      <ModalConfirmCancel
        isOpen={true}
        headerText={'header'}
        bodyText={'body'}
        confirmButtonText={'confirm'}
        cancelButtonText={'cancel'}
        confirmAction={confirmAction}
        cancelAction={cancelAction}
      />
    );
    fireEvent.click(screen.getByText('cancel'));
    expect(confirmAction.mock.calls.length).toEqual(0);
    expect(cancelAction.mock.calls.length).toEqual(1);
  });
});
