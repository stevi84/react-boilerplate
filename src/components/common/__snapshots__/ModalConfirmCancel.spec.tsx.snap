// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModalConfirmCancel > should equal saved snapshot 1`] = `
<ForwardRef(Dialog)
  id="modal-confirm-cancel"
  onClose={[Function]}
  open={true}
>
  <ForwardRef(DialogTitle)>
    header
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(DialogContentText)>
      body
    </ForwardRef(DialogContentText)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Button)
      autoFocus={true}
      color="primary"
      id="modal-confirm-cancel-button-confirm"
      onClick={[Function]}
      variant="contained"
    >
      confirm
    </ForwardRef(Button)>
    <ForwardRef(Button)
      color="primary"
      id="modal-confirm-cancel-button-cancel"
      onClick={[Function]}
      variant="outlined"
    >
      cancel
    </ForwardRef(Button)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
